system/jlib/jsocket.cpp

```diff
@@ -1426,14 +1442,31 @@ void CSocket::connect_wait(unsigned timems)
             CriticalBlock block(crit);
             --connectingcount;
         }
-        if (err==0) {
+        if (err==0)
+        {
             err = post_connect();
-            if (err==0) {
+            if (err==0)
+            {
                 STATS.connects++;
                 STATS.connecttime+=usTick()-startt;
 #ifdef _TRACE
                 setTraceName();
 #endif              
+#ifdef SOCKTRACE
+                unsigned conn_mstime = (usTick() - startt) / 1000;
+                int my_loc_port = 0;
+                struct sockaddr_in sin1;
+                socklen_t len1 = sizeof(sin1);
+                if (getsockname(sock, (struct sockaddr *)&sin1, &len1) == 0)
+                    my_loc_port = ntohs(sin1.sin_port);
+                int my_rem_port = 0;
+                struct sockaddr_in sin2;
+                socklen_t len2 = sizeof(sin2);
+                if (getpeername(sock, (struct sockaddr *)&sin2, &len2) == 0)
+                    my_rem_port = ntohs(sin2.sin_port);
+                PROGLOG("SOCKTRACE connect_wait(%u): ok - %u ms fd: %d loc port: %d rem port: %d", timems, conn_mstime, sock, my_loc_port, my_rem_port);
+                // PrintStackReport();
```

Comment:
Is this exactly the same as code above (starting at line 1284) apart from the text connect_wait vs connect_timeout ?
If so, should common up.